nature-logger/Controllers/PlantsViewControllers.swift

```diff
@@ -0,0 +1,149 @@
+            let common_name = heros[indexPath.row].common_name
+            cell.plantTitle.text = common_name == nil ? "Unkown name" : heros[indexPath.row].common_name.capitalized
+            return cell
+        }else{
+            let cell = tableView.dequeueReusableCell(withIdentifier: "loadingCell", for: indexPath) as! LoadingCell
+            cell.spinner.startAnimating()
+            return cell
+        }
+    }
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 100
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        performSegue(withIdentifier: "showDetails", sender: self)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let destination = segue.destination as? PlantsDetailedViewController {
+            destination.hero = heros[(tableView.indexPathForSelectedRow?.row)!]
+        }
+    }
+    
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let offsetY = scrollView.contentOffset.y
+        let contentHeight = scrollView.contentSize.height
+        
+        if offsetY > contentHeight - scrollView.frame.height {
+            if !fetchingMore {
+                beginBatchFetch()
+            }
+        }
+        
+    }
+    
+    func beginBatchFetch() {
+        fetchingMore = true;
+        print("beginBatchFetch")
+        tableView.reloadSections(IndexSet(integer: 1), with: .none)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0, execute: {
+            self.downloadJSON {
+                self.fetchingMore = false
+                self.tableView.reloadData()
+            }
+        })
+    }
+    
+    // Needs to fix space search issue
+
+    func downloadJSON(completed: @escaping () -> ()){
+        var urlString = "https://trefle.io/api/v1/plants/?token=sqvdZFYkPO5aF0BmIMYoJAkU5hCNM7-uzXdhTmc3row&page=1"
+        
+        if(searchString.isEmpty == false){
+            urlString = "https://trefle.io/api/v1/plants/search?token=sqvdZFYkPO5aF0BmIMYoJAkU5hCNM7-uzXdhTmc3row&page=1&q=\(self.searchString)"
+        }
+        
+        print(urlString)
+
+        let url = URL(string: urlString)
+        
+        URLSession.shared.dataTask(with: url!){ (data,response, error) in
+            if error == nil {
+                do {
+                    let jsonData = try JSONDecoder().decode(Plants.self, from: data!)
+                    
+                    if(self.page == 1){
+                        self.heros = jsonData.data
+                    }else{
+                        self.heros.append(contentsOf: jsonData.data)
+                    }
+                    self.page = self.page + 1
+
+                    DispatchQueue.main.async {
+                        completed()
+                    }
+                    
+                }catch{
+                    print("JSON Error", response)
```

Comment:
Would be very nice if you showed a popup here that the data could not be loaded :smiley: